azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1729,6 +1727,28 @@ private void cleanExecutionLogs() {
     }
   }
 
+  /**
+   * Calls executor to dispatch the flow, update db to assign the executor and
+   * in-memory state of executableFlow
+   */
+  private void dispatch(ExecutionReference reference, ExecutableFlow exflow,
+    Executor choosenExecutor) throws ExecutorManagerException {
+    exflow.setUpdateTime(System.currentTimeMillis());
+    callExecutorServer(exflow, choosenExecutor,
+      ConnectorParams.EXECUTE_ACTION);
+    executorLoader.assignExecutor(choosenExecutor.getId(),
```

Comment:
I am just wondering: what happens if this sql command does not complete and throws an exception?
The system will be in an inconsistent state of having submitted the flow, but not having it registered in the DB (and not be in the RunningFlows map, etc).  Should we handle this scenario?